### PR TITLE
PD-1707 Refresh button

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

**Review by shaneeza — CHANGES_REQUESTED** [review]
Looking good! A few things:

- The side padding for each size is a few pixels off.
- It looks like for the icon only button, the icon fill should change on hover and focus depending on the variant.
- For `Primary Outline` it looks like the background color should change on focus and hover.

I'm using this [design](https://www.figma.com/file/4h2VwjCuJUbeZ7hzD2J1rq/branch/EafsSYjAX3VYeY832iLiR9/LeafyGreen-Design-System?node-id=4444%3A19384)

File: packages/button/src/ButtonIcon.tsx [hunk 1]
```diff
@@ -90,6 +91,13 @@ const onlyIconColor: Record<Mode, Record<Variant, string>> = {
   },
 };
 
+const onlyIconStyleHover = css`
+  ${ButtonDataProp.selector}:hover &,
```
Comment (shaneeza) [inline packages/button/src/ButtonIcon.tsx:95]: Out of curiosity, is this a way to target the parent?

**Review by shaneeza — APPROVED** [review]
✨

File: packages/button/src/ButtonIcon.tsx (hunk 1 above)
Comment (TheSonOfThomp) [reply packages/button/src/ButtonIcon.tsx:95]: Unfortunately, yes. The other way to do it would be by switching a css variable in the parent styles

**Pull request merged**